utils/buffer/reader.go

```diff
@@ -0,0 +1,316 @@
+package buffer
+
+import (
+	"encoding/binary"
+	"fmt"
+	"unsafe"
+)
+
+// ReadAsUint64 reads an uint64 from r and stores the result into c with pointer type casting into type T.
+func ReadAsUint64[T any](r Reader, c *T) (n int64, err error) {
+	/* #nosec G103 -- behavior and consequences well understood */
```

Comment:
Let's just say briefly why it's understood

Reply:
Done